python/ray/data/read_api.py

```diff
@@ -3011,6 +3013,89 @@ def from_torch(
     )
 
 
+@PublicAPI
+def read_iceberg(
+    *,
+    table_identifier: str,
+    row_filter: Union[str, "BooleanExpression"] = None,
+    parallelism: int = -1,
+    selected_fields: Tuple[str, ...] = ("*",),
+    snapshot_id: Optional[int] = None,
+    scan_kwargs: Optional[Dict[str, str]] = None,
+    catalog_kwargs: Optional[Dict[str, str]] = None,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+    override_num_blocks: Optional[int] = None,
+) -> Dataset:
+    """Create a :class:`~ray.data.Dataset` from an Iceberg table.
+
+    The table to read from is specified using a fully qualified ``table_identifier``.
+    Using PyIceberg, any intended row filters, snapshot IDs, etc. are applied,
```

Comment:
nit: we follow the google style guide for docs, which recommends avoiding use of "etc" and other abbreviations. could you enumerate the full list of items that are applied, or reword otherwise? (you could also link to specific pyiceberg documentation for more details, like you included for `catalog_kwargs`) 

https://developers.google.com/style/word-list#etc